learn/what_is_meilisearch/comparison_to_alternatives.md

```diff
@@ -120,7 +120,7 @@ Can't find a client you'd like us to support? [Submit your idea or vote for it](
 | High availability | Available with [Meilisearch Cloud](https://cloud.meilisearch.com/register) | ✅ | ✅ | ✅ |
 | Run-time dependencies | None | N/A | None | None |
 | Backward compatibility | ✅ | N/A | ✅ | ✅ |
-| Upgrade path | Documents need to be reindexed | N/A  | Documents need to be reindexed | Documents need to be reindexed |
+| Upgrade path | Documents need to be reindexed | N/A  | Documents need not be reindexed  | Documents need to be reindexed |
```

Comment:
This has always been the case with Typesense. Users just replace binary and restart the process. Documents need not be sent over again via the API.

Reply:
Reindexing in our documentation means that the engine processes the documents to compute optimized data structure to search in them efficiently. Therefore, Typesense always reindexes the documents when the binary is updated and even every time the binary is started. Am I right?

Reply:
By your definition of reindexing, meaning that the engine is the one that's doing the indexing, that's right - Typesense does reindex automatically on version upgrades, internally.

In our documentation, we refer to reindexing as the process of users sending documents via the API.

Reply:
Is that also true that, [according to our definition of indexing](https://en.wikipedia.org/wiki/Search_engine_indexing), Typesense does re-index the documents when is started?

Reply:
As we're in a Meilisearch website, it makes sense to use Meilisearch definition of reindexing.

Reply:
> In our documentation, we refer to reindexing as the process of users sending documents via the API.

So, why did you write that we need that in you comparison table, the user doesn't need to send the documents via the API. Can you remove this sentence, please?

> Version upgrades require re-indexing.

Reply:
> Is that also true that, [according to our definition of indexing](https://en.wikipedia.org/wiki/Search_engine_indexing), Typesense does re-index the documents every time it is started/run as the data-structures are in memory?

That's correct. But Typesense does this automatically internally.

> So, why did you write that we need that in the comparison table of Typesense? Users don't need to send the documents via the API. Can you remove the following sentence, please?
> 
> > Version upgrades require re-indexing.

I just read your docs [here](https://docs.meilisearch.com/learn/advanced/snapshots_vs_dumps.html#dumps) and realized that I've been thinking about your "snapshots" feature all this while. I didn't realize you launched the "dumps" feature which is compatible across version. Good to know, I'll update our table. 

> Could you also update the Elastic Search part about backward compatibility, [they are not](https://discuss.elastic.co/t/is-is-possible-to-run-elasticsearch-6-in-a-backward-compatible-mode/113211/3).

Good to know. I'll update.

> BTW, where did you get the information about the internal Algolia index backward compatibility? There is no information about that on the internet.

Since they operate as a SaaS service, they most likely manage backward/forward compatibility internally and don't expose it to users. Since users don't have to worry about their internal index versions, I marked them as "Fully backward compatible" at least from an end user's perspective.

Reply:
Thank you very much Jason.

BTW dumps are available since nearly the beginning. Maybe pre-v0.20, so at least Mar 30 2021.

You should probably put _N/A_ for Algolia, as the category specifically talk about a technical aspect that we can't verify. What do you think?

Reply:
> You should probably put N/A for Algolia, as the category specifically talk about a technical aspect that we can't verify. What do you think?

I was just thinking the same! I'll add a note that they do this internally automatically.

Reply:
Just found this in the [Elasticsearch docs](https://www.elastic.co/guide/en/elasticsearch/reference/current/setup-upgrade.html#upgrade-index-compatibility):

> Elasticsearch has full query and write support for indices created in the previous major version. If you have indices created in 6.x or earlier, you might use the [archive functionality](https://www.elastic.co/guide/en/elasticsearch/reference/current/archive-indices.html) to import them into newer Elasticsearch versions, or you must reindex or delete them before upgrading to 8.6.2

So they are backward compatible up to 1 major version.

Reply:
About the original comment again. Could you please update the sentence to something like: « documents will be automatically reindexed » for Meilisearch, Typesense and ES, please?

Reply:
Done.